mistakedetection.tests/src/ca/mcgill/sel/mistakedetection/tests/MistakeDetectionPerformanceAnalysis.java

```diff
@@ -36,387 +67,235 @@
-
-    var comparison = MistakeDetection.compare(instructorSolution, studentSolution, true);
-
+  /** Tests that the MDS runs without errors on the submissions from the final exam dataset. */
+  @Test public void testThatMdsRunsOnFinalExamStudentSolutions() {
+    var ma = maf.createModelingAssistant();
+    Map<Integer, List<Integer>> solutionsToNumMistakes = new TreeMap<>();
+    var instSolutions = setupSmartHomeInstructorSolutions();
+    for (var instSolution: instSolutions) {
+      try (var files = Files.walk(Path.of(FINAL_EXAM_SUBMISSIONS_PATH), 2)) {
+        List<String> validCdms = new ArrayList<>();
+        List<String> invalidCdms = new ArrayList<>();
+        var filenamePattern = Pattern.compile("\\d+\\.cdm");
+        var studentCdms = files
+            .filter(file ->
+                !Files.isDirectory(file) && filenamePattern.matcher(file.getFileName().toString()).matches()
+                && !file.endsWith("0.cdm"))
+            .map(file -> {
+              var cdm = ResourceHelper.cdmFromFile(file);
+              cdm.setName(file.getFileName().toString().replace(".cdm", ""));
+              return cdm;
+            });
+
+        studentCdms.forEach(cdm -> {
+          var cdmName = cdm.getName();
+          var spreadsheetName = cdmName + "_" + instSolution.getClassDiagram().getName();
+          var student = maf.createStudent();
+          student.setModelingAssistant(ma);
+          student.setName("Student" + cdmName); // Student1 and so on
+          var studSolution = maf.createSolution();
+          studSolution.setModelingAssistant(ma);
+          studSolution.setStudent(student);
+          studSolution.setClassDiagram(cdm);
+          try {
+            System.out.println("\n\nDetecting mistakes for " + cdmName);
+            var comparison = applyFinalExamCriteria(MistakeDetection.compare(instSolution, studSolution))
+                .logUnmappedItemsOnly();
+            assertNotNull(comparison);
+            if (OVERWRITE_SPREADSHEETS) {
+              produceExcelSheet(comparison, spreadsheetName, SMART_HOME_OUTPUT_SPREADSHEET_LOC + "/" + cdmName);
+            }
+            var solutionNum = Integer.parseInt(cdmName);
+            if (solutionsToNumMistakes.containsKey(solutionNum)) {
+              solutionsToNumMistakes.get(solutionNum).add(comparison.newMistakes.size());
+            } else {
+              solutionsToNumMistakes.put(solutionNum, new ArrayList<>(List.of(comparison.newMistakes.size())));
+            }
+            if (!invalidCdms.contains(cdmName)) {
+              validCdms.add(cdmName);
+            }
+          } catch (Exception e) {
+            System.err.println("Could not detect mistakes for " + cdmName + " due to error:");
+            e.printStackTrace();
+            Comparison.instances.remove(Comparison.instances.size() - 1); // MDS compare failed -> don't track instance
+            invalidCdms.add(cdmName);
```

Comment:
Some of the submitted student solutions were invalid so they were not considered in our analysis